src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,693 @@
+
+        // Initialize the last available assets to the current available assets
+        // This ensures no performance fee is accrued when the performance fee is calculated when the fee is set
+        lastAvailableAssets = SafeCast.toUint128(_availableAssets());
+        _setFee(_fee);
+        _setFeeCollector(_feeCollector);
+
+        liquidityProviderController = _liquidityProviderController;
+        emit LiquidityProviderControllerUpdated(_liquidityProviderController);
+    }
+
+    ////////////////////////////////////////////////////
+    ///                 Swap Functions
+    ////////////////////////////////////////////////////
+
+    /**
+     * @notice Swaps an exact amount of input tokens for as many output tokens as possible.
+     * msg.sender should have already given the ARM contract an allowance of
+     * at least amountIn on the input token.
+     *
+     * @param inToken Input token.
+     * @param outToken Output token.
+     * @param amountIn The amount of input tokens to send.
+     * @param amountOutMin The minimum amount of output tokens that must be received for the transaction not to revert.
+     * @param to Recipient of the output tokens.
+     */
+    function swapExactTokensForTokens(
+        IERC20 inToken,
+        IERC20 outToken,
+        uint256 amountIn,
+        uint256 amountOutMin,
+        address to
+    ) external virtual {
+        uint256 amountOut = _swapExactTokensForTokens(inToken, outToken, amountIn, to);
+        require(amountOut >= amountOutMin, "ARM: Insufficient output amount");
+    }
+
+    /**
+     * @notice Uniswap V2 Router compatible interface. Swaps an exact amount of
+     * input tokens for as many output tokens as possible.
+     * msg.sender should have already given the ARM contract an allowance of
+     * at least amountIn on the input token.
+     *
+     * @param amountIn The amount of input tokens to send.
+     * @param amountOutMin The minimum amount of output tokens that must be received for the transaction not to revert.
+     * @param path The input and output token addresses.
+     * @param to Recipient of the output tokens.
+     * @param deadline Unix timestamp after which the transaction will revert.
+     * @return amounts The input and output token amounts.
+     */
+    function swapExactTokensForTokens(
+        uint256 amountIn,
+        uint256 amountOutMin,
+        address[] calldata path,
+        address to,
+        uint256 deadline
+    ) external virtual returns (uint256[] memory amounts) {
+        require(path.length == 2, "ARM: Invalid path length");
+        _inDeadline(deadline);
+
+        IERC20 inToken = IERC20(path[0]);
+        IERC20 outToken = IERC20(path[1]);
+
+        uint256 amountOut = _swapExactTokensForTokens(inToken, outToken, amountIn, to);
+
+        require(amountOut >= amountOutMin, "ARM: Insufficient output amount");
+
+        amounts = new uint256[](2);
+        amounts[0] = amountIn;
+        amounts[1] = amountOut;
```

Comment:
I'm asking it here because I don't know if it's a mistake or something deliberate for the integrator.
The value of `amountOut` might be wrong when the output token is `steth` as we add `2` extra wei.

Is it done on purpose to ensure the integrator has at least the amount of steth returned?

Reply:
That's an excellent point. We do add an extra 2 wei but that doesn't always get received by the swapper. For larger swaps that will only receive the smaller `amountOut`. For smaller swaps they will receive `amountOut` + 1 or if the rounding works in their favour, they might receive `amountOut` + 2.

Reply:
The extra 2 wei was mainly added for `swapTokensForExactTokens` where the swapper specifies exactly how much stETH they want to receive. I've added the extra 2 wei so they at least get what they specified but may get a little more.

I'd be interesting to see what other Uniswap like pools with stETH do. I suspect the swapper receives 1-2 wei less than what they specified when receiving stETH with `swapTokensForExactTokens`.

Reply:
Ok thanks for the explanation! 
I'll try to investigate, but as I remember they receive 1-2 wei less.

Reply:
I'm tempted to just remove the logic that sends an extra 2 wei to swappers so they get at least what they requested on a `swapTokensForExactTokens`.

@DanielVF what do you think?

```
/**
 * @dev Due to internal stETH mechanics required for rebasing support, in most cases stETH transfers are performed
 * for the value of 1 wei less than passed to transfer method. Larger transfer amounts can be 2 wei less.
 *
 * The MultiLP implementation ensures any WETH reserved for the withdrawal queue is not used in swaps from stETH to WETH.
 */
function _transferAsset(address asset, address to, uint256 amount) internal override {
    // Add 2 wei if transferring stETH
    uint256 transferAmount = asset == address(token1) ? amount + 2 : amount;

    super._transferAsset(asset, to, transferAmount);
}
```